README.md
# flutter_ds_bfi

Design System Package for BFI

## Getting Started

This project is a starting point for a Dart
[package](https://flutter.dev/developing-packages/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://flutter.dev/docs), which offers tutorials, 
samples, guidance on mobile development, and a full API reference.

## Example for DSText & FontFamily
![DSText & FontFamily](example/screenshot/fontfamily.png)

### How to use ?
Just add this syntax to used DSText
```dart
DSText(
                  data: 'Font Family - Poppins',
                  textStyle: TextStyle(
                    fontFamily: DSFont.Poppins,
                    color: Colors.black,
                  ),
                )
```
And this is the stlye
```dart
class DSTextStyle {
  static const TextStyle defaultStyle = TextStyle(
      fontSize: 14.0,
      fontFamily: DSFont.Poppins,
      color: Colors.black,
      fontWeight: FontWeight.normal);

  static const TextStyle regularStyle = TextStyle(
      fontSize: 12.0,
      fontFamily: DSFont.Poppins,
      color: Colors.black,
      fontWeight: FontWeight.normal);

  static const TextStyle titleTextStyle = TextStyle(
      fontSize: 21.0,
      fontFamily: DSFont.Poppins,
      color: Colors.black,
      fontWeight: FontWeight.bold);
}
```

## Example for DSColor
![DSColor](example/screenshot/color.png)

### How to use ?
Just add this syntax to used DSColor
```dart
buildTileColor(
                    "Primary Color", DSColor.primaryBlue, Colors.white),
```
And this is list of Color
```dart
Class DSColor {
  DSColor._();

  static const Color primaryBlue = Color(0xFF04559f);
  static const Color bgColor = Color(0xFFf7f8fa);
  static const Color successGreen = Color(0xFF7ac143);
  static const Color secondaryOrange = Color(0xFFd47c0a);
  static const Color disableColor = Color(0xFFA9A9A9);
}
```

## Example for DSButton
![DSButton](example/screenshot/button.png)

### How to use ?
Just add this syntax to used DSButton. Example Active Button 
```dart
DSFilledButton(
                  text: "Active Button",
                  fontSize: 16,
                  buttonState: DSButtonState.Active,
                  fontWeight: FontWeight.normal,
                  onTap: () {},
                )
```
And this is list of Button State
```dart
enum DSButtonState { Active, Disable, Tapable, Ghost, GhostDisable }
```

## Example for DSCard
![DSButton](example/screenshot/card.png)

### How to use ?
Just add this syntax to used DSCard. Example Blue Card 
```dart
DSCard(
              angsuranPerBulan: Text(
                'Rp. 500.000',
                style: TextStyle(
                  fontSize: 14,
                  color: Colors.white,
                  fontWeight: FontWeight.bold,
                ),
              ),
              jatuhTempo: Text('12 Desember 2020',
                  style: TextStyle(
                      fontSize: 14,
                      color: Colors.white,
                      fontWeight: FontWeight.bold)),
              licensePlate: 'B1MO',
              noKontrak: '054651064130',
              bgCard: 'assets/imgs/card_blue.png',
              icon: 'assets/imgs/mobil.png',
              titleAngsuran: 'Tagihan Bulan Ini',
              arrowUp: 'assets/imgs/arrow_up_white.png',
              onTap: () {
                print('tap tap tap');
              },
            )
```
Just add this syntax to used DSCard. Example Contact Us 
```dart
  DSCardContactUs(
                image: 'assets/imgs/phone_call.png',
                title: 'Telepon Kami',
                subtitle: 'Senin - Jumat: 08.00-18.00 WIB',
                subtitle2: 'Sabtu: 08.00 - 12.00 WIB')
```




CHANGELOG.md
## [0.0.1] - 24 November 2020.

First Publish Design System to Pub Dev

## [0.0.1+1] - 24 November 2020.

Update Card Padding

## [0.0.1+2] - 25 November 2020.

Add Custom Card for Contact Us

## [0.0.1+3] - 01 December 2020.

Add new param titleAngsuran on DSCard

## [0.0.1+4] - 01 December 2020.

Add new param subtitle2 on DSCardContactUs

## [0.0.1+5] - 01 December 2020.

Add logic in three lines on DSCardContactUs

## [0.0.1+6] - 01 December 2020.

Bug Fix on DSCardContactUs & DSCard

## [0.0.1+7] - 02 December 2020.

Bug Fix on DSCardContactUs

## [0.0.1+8] - 04 December 2020.

Bug Fix on DSCardContactUs

## [0.0.1+9] - 08 December 2020.

Bug Fix on DSCardContactUs shadow

## [0.0.1+10] - 22 December 2020.

Bug Fix on DSCard. Add 2 param, onTap and Icon Arrow

## [0.0.1+11] - 22 December 2020.

Bug Fix on DSCardContactUs Padding

## [0.0.1+12] - 14 January 2021.

Happy New Year 2021

## [0.0.1+13] - 14 January 2021.

Delete icon arrow DSCard

## [0.0.1+14] - 21 January 2021.

Change String angsuran perbulan & jatuh tempo into widget on DSCard

## [0.0.1+15] - 16 February 2021.

Add General DS Button, Add Card for Product Information

## [0.0.1+16] - 16 February 2021.

Change Image Asset to Network on Card Product Information

## [0.0.1+17] - 16 February 2021.

Add font color for General DS Button

## [0.0.1+18] - 17 February 2021.

Add General DS Ghost Button

## [0.0.1+20] - 20 December 2021.

Add General DS List Text Field

## [0.0.1+21] - 20 December 2021.

Fixing DS List Text Field

## [0.0.1+22] - 30 December 2021.

Fixing DS Custom Text Field

## [0.0.1+23] - 03 January 2022.

Fixing DS Custom Text Field Parameter

## [0.0.1+24] - 11 January 2022.

Add DS Card Label

## [0.0.1+25] - 11 February 2022.

Add New DS Card Label

## [0.0.1+26] - 11 February 2022.

Add New DS Card Label, Fixing Bug



example/lib/component_list/card_screen.dart
import 'package:flutter_ds_bfi/flutter_ds_bfi.dart';
import 'package:flutter/material.dart';

class CardScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Card'),
        centerTitle: true,
      ),
      body: SingleChildScrollView(
          child: Column(
        children: [
          Padding(
            padding: const EdgeInsets.only(left: 18.0, right: 18.0, top: 18.0),
            child: DSCard(
              angsuranPerBulan: Text(
                'Rp. 500.000',
                style: TextStyle(
                  fontSize: 14,
                  color: Colors.white,
                  fontWeight: FontWeight.bold,
                ),
              ),
              jatuhTempo: Text('12 Desember 2020',
                  style: TextStyle(
                      fontSize: 14,
                      color: Colors.white,
                      fontWeight: FontWeight.bold)),
              licensePlate: 'B1MO',
              noKontrak: '054651064130',
              bgCard: 'assets/imgs/card_blue.png',
              icon: 'assets/imgs/mobil.png',
              titleAngsuran: 'Tagihan Bulan Ini',
              arrowUp: 'assets/imgs/arrow_up_white.png',
              onTap: () {
                print('tap tap tap');
              },
            ),
          ),
          SizedBox(height: 30),
          Padding(
            padding: const EdgeInsets.only(left: 18.0, right: 18.0),
            child: DSCardContactUs(
                image: 'assets/imgs/phone_call.png',
                title: 'Telepon Kami',
                subtitle: 'Senin - Jumat: 08.00-18.00 WIB',
                subtitle2: 'Sabtu: 08.00 - 12.00 WIB'),
          ),
          SizedBox(height: 30),
          Padding(
            padding: const EdgeInsets.only(left: 18.0, right: 18.0),
            child: DSCardContactUs(
                image: 'assets/imgs/phone_call.png',
                title: 'Telepon Kami',
                subtitle: 'Senin - Jumat: 08.00-18.00 WIB'),
          ),
          SizedBox(height: 30),
          Container(
              padding: const EdgeInsets.only(left: 8.0, right: 8.0),
              width: double.infinity,
              height: MediaQuery.of(context).size.height * 0.21,
              child: ListView(
                padding: EdgeInsets.all(8),
                scrollDirection: Axis.horizontal,
                children: [
                  DSCardLabel(
                    width: MediaQuery.of(context).size.width * 0.68,
                    height: MediaQuery.of(context).size.height * 0.21,
                    label: 'Pending',
                    title: 'Reza Rahardian',
                    header: Padding(
                        padding: const EdgeInsets.only(left: 16.0),
                        child: Row(
                          children: [
                            DSText(
                              data: 'Agreement No : ',
                              textStyle: TextStyle(
                                color: Colors.black,
                                fontSize: 14,
                              ),
                            ),
                            DSText(
                              data: '40117201022',
                              textStyle: TextStyle(
                                color: DSColor.primaryBlue,
                                fontSize: 14,
                              ),
                            ),
                          ],
                        )),
                    content: Padding(
                        padding: const EdgeInsets.only(left: 16.0),
                        child: Row(
                          children: [
                            DSText(
                              data: 'Applicant ID : ',
                              textStyle: TextStyle(
                                color: Colors.black,
                                fontSize: 14,
                              ),
                            ),
                            DSText(
                              data: '201719919293810AS',
                              textStyle: TextStyle(
                                color: Colors.black,
                                fontSize: 14,
                              ),
                            ),
                          ],
                        )),
                    footer: Padding(
                      padding: const EdgeInsets.only(left: 16.0),
                      child: DSText(
                          data: '24/04/2018 11:42',
                          textStyle:
                              TextStyle(color: Colors.grey, fontSize: 14)),
                    ),
                  ),
                  SizedBox(width: 16.0),
                  DSCardLabel(
                    width: MediaQuery.of(context).size.width * 0.68,
                    height: MediaQuery.of(context).size.height * 0.21,
                    label: 'Pending',
                    title: 'Reza Rahardian',
                    header: Padding(
                        padding: const EdgeInsets.only(left: 16.0),
                        child: Row(
                          children: [
                            DSText(
                              data: 'Agreement No : ',
                              textStyle: TextStyle(
                                color: Colors.black,
                                fontSize: 14,
                              ),
                            ),
                            DSText(
                              data: '40117201022',
                              textStyle: TextStyle(
                                color: DSColor.primaryBlue,
                                fontSize: 14,
                              ),
                            ),
                          ],
                        )),
                    content: Padding(
                        padding: const EdgeInsets.only(left: 16.0),
                        child: Row(
                          children: [
                            DSText(
                              data: 'Applicant ID : ',
                              textStyle: TextStyle(
                                color: Colors.black,
                                fontSize: 14,
                              ),
                            ),
                            DSText(
                              data: '201719919293810AS',
                              textStyle: TextStyle(
                                color: Colors.black,
                                fontSize: 14,
                              ),
                            ),
                          ],
                        )),
                    footer: Padding(
                      padding: const EdgeInsets.only(left: 16.0),
                      child: DSText(
                          data: '24/04/2018 11:42',
                          textStyle:
                              TextStyle(color: Colors.grey, fontSize: 14)),
                    ),
                  ),
                ],
              )),
          Padding(
            padding: const EdgeInsets.only(left: 18.0, right: 18.0, top: 18.0),
            child: DSCardLabel(
              width: MediaQuery.of(context).size.width,
              height: MediaQuery.of(context).size.height * 0.21,
              label: 'Submitted',
              title: 'Reza Rahardian',
              header: Padding(
                  padding: const EdgeInsets.only(left: 16.0),
                  child: Row(
                    children: [
                      DSText(
                        data: 'Agreement No : ',
                        textStyle: TextStyle(
                          color: Colors.black,
                          fontSize: 14,
                        ),
                      ),
                      DSText(
                        data: '40117201022',
                        textStyle: TextStyle(
                          color: DSColor.primaryBlue,
                          fontSize: 14,
                        ),
                      ),
                    ],
                  )),
              content: Padding(
                  padding: const EdgeInsets.only(left: 16.0),
                  child: Row(
                    children: [
                      DSText(
                        data: 'Applicant ID : ',
                        textStyle: TextStyle(
                          color: Colors.black,
                          fontSize: 14,
                        ),
                      ),
                      DSText(
                        data: '201719919293810AS',
                        textStyle: TextStyle(
                          color: Colors.black,
                          fontSize: 14,
                        ),
                      ),
                    ],
                  )),
              footer: Padding(
                padding: const EdgeInsets.only(left: 16.0),
                child: DSText(
                    data: '24/04/2018 11:42',
                    textStyle: TextStyle(color: Colors.grey, fontSize: 14)),
              ),
            ),
          ),
        ],
      )),
    );
  }
}
